src/Aspire.Hosting.Yarp/ConfigurationBuilder/IYarpConfigurationBuilder.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Yarp;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Interface to build a configuration file for YARP
+/// </summary>
+public interface IYarpConfigurationBuilder
+{
+    /// <summary>
+    /// Add a new route to YARP that will target the cluster in parameter.
+    /// </summary>
+    /// <param name="path">The path to match for this route.</param>
+    /// <param name="cluster">The target cluster for this route.</param>
+    /// <returns></returns>
+    public YarpRoute AddRoute(string path, YarpCluster cluster);
+
+    /// <summary>
+    /// Add a new cluster to YARP.
+    /// </summary>
+    /// <param name="endpoint">The endpoint used by this cluster.</param>
+    /// <returns></returns>
+    public YarpCluster AddDestination(EndpointReference endpoint);
+}
+
+/// <summary>
+/// Collection of extensions methods for <see cref="IYarpConfigurationBuilder"/>
+/// </summary>
+public static class YarpConfigurationBuilderExtensions
+{
+    private const string CatchAllPath = "/{**catchall}";
+
+    /// <summary>
+    /// Add a new catch all route to YARP that will target the cluster in parameter.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="cluster">The target cluster for this route.</param>
+    /// <returns></returns>
+    public static YarpRoute AddRoute(this IYarpConfigurationBuilder builder, YarpCluster cluster)
+    {
+        return builder.AddRoute(CatchAllPath, cluster);
+    }
+
+    /// <summary>
+    /// Add a new route to YARP that will target the cluster in parameter.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="path">The path to match for this route.</param>
+    /// <param name="endpoint">The target endpoint for this route.</param>
+    /// <returns></returns>
+    public static YarpRoute AddRoute(this IYarpConfigurationBuilder builder, string path, EndpointReference endpoint)
+    {
+        var cluster = builder.AddDestination(endpoint);
```

Comment:
Looking at this now I think that having a method called `AddDestination` that returns a `YarpCluster` could be confusing. Given that we have the other `AddRoute` overloads that take an endpoint, maybe we should just name the method `AddCluster` for Yarp terminology alignment purposes.

(am I driving you crazy yet? :))

Reply:
Ha it's a relic from my previous proto, I will remove it, it's not needed with the current design

Reply:
Ha it's a relic of a previous proto, yes you are right it should be `AddCluster`, I missed it during the rename.